code/modules/research/server.dm

```diff
@@ -59,8 +57,9 @@
 			heat_health = min(100, heat_health + 1)
 		if(T0C to (T20C + 20))
 			heat_health = Clamp(heat_health, 0, 100)
-		if((T20C + 20) to (T0C + 70))
-			heat_health = max(0, heat_health - 1)
+		if((T20C + 20) to INFINITY)
+			var/excess = environment.temperature - (T20C + 20)
+			heat_health = max(0, -(excess/40))
```

Comment:
This max will always return 0, thus is pointless.


But even beyond that, the whole switch statement here is confusing and irrelevant.
The amount of overheating doesn't change how much the server loses, it's either overheating or it isn't.

Consider the following:

```
 	switch(environment.temperature)
 		if(0 to T0C)
  			overheating = FALSE
 		if((T20C + 20) to INFINITY)
 			overheating = TRUE
	if(overheating)
		//bad_stuff
```

With the way it currently is written, that is identical, and is MUCH easier to read.

Or an alternative method:
```
 	switch(environment.temperature)
 		if(0 to T0C)
  			heat_health++
 		if((T20C + 20) to INFINITY)
			var/excess = environment.temperature - (T20C + 20)
 			heat_health -= excess/40
	heat_health = Clamp(heat_health, 0, 100)
	if(heat_health <= 0)
		//bad_stuff
```

With this second method, the server will have a short grace period when overheating before it starts losing R&D levels.